### PR TITLE
Store task execution event reason & task type

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Store task execution event reason and task type

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/repositories/transformers/task_execution.go [hunk 1]
```diff
@@ -209,6 +210,9 @@ func UpdateTaskExecutionModel(request *admin.TaskExecutionEventRequest, taskExec
 	taskExecutionClosure.Phase = request.Event.Phase
 	taskExecutionClosure.UpdatedAt = request.Event.OccurredAt
 	taskExecutionClosure.Logs = mergeLogs(taskExecutionClosure.Logs, request.Event.Logs)
+	if len(request.Event.Reason) > 0 {
```
Comment (kumare3) [inline pkg/repositories/transformers/task_execution.go:213]: what about task type?

**Review by pmahindrakar-oss — APPROVED** [review]

File: pkg/repositories/transformers/task_execution_test.go [hunk 2]
```diff
@@ -151,6 +151,8 @@ func TestCreateTaskExecutionModelQueued(t *testing.T) {
 				RetryAttempt:          1,
 				InputUri:              "input uri",
 				OccurredAt:            taskEventOccurredAtProto,
+				Reason:                "Task was scheduled",
+				TaskType:              "dynamic",
```
Comment (kumare3) [inline pkg/repositories/transformers/task_execution_test.go:155]: there is no task called dynamic right?

File: pkg/repositories/transformers/task_execution.go [hunk 3]
```diff
@@ -209,6 +211,9 @@ func UpdateTaskExecutionModel(request *admin.TaskExecutionEventRequest, taskExec
...
+		taskExecutionClosure.Reason = request.Event.Reason
```
Comment (pmahindrakar-oss) [inline pkg/repositories/transformers/task_execution.go:215]: Can we add a unit test to verify this gets added in the closure from the event along with the task type

File: pkg/repositories/transformers/task_execution_test.go (hunk 2 above)
Comment (katrogan) [reply pkg/repositories/transformers/task_execution_test.go:155]: this is just a unit test value

File: pkg/repositories/transformers/task_execution.go (hunk 3 above)
Comment (katrogan) [reply pkg/repositories/transformers/task_execution.go:215]: yup already done as part of this PR

**Comment by katrogan** [comment]
ptal @pmahindrakar-oss @kumare3

**Review by pmahindrakar-oss — APPROVED** [review]
Probably these two are also coming as empty
startedAt https://github.com/flyteorg/flyteadmin/blob/ab2087db162412e80a690f09c701aa6da8764d78/pkg/repositories/transformers/task_execution.go#L36

And 
duration https://github.com/flyteorg/flyteadmin/blob/ab2087db162412e80a690f09c701aa6da8764d78/pkg/repositories/transformers/task_execution.go#L57

**Comment by katrogan** [comment]
@pmahindrakar-oss sure, I'll save that fix for propeller :)

**Pull request merged**